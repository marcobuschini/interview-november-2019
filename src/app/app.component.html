
<parking-widget #widget [vendor]="vendor"></parking-widget>