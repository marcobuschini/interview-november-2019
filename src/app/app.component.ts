import { Component, Input } from '@angular/core';

import { WidgetComponent, Vendor } from 'widget';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  title = 'atSistemas-interview';
  vendor: Vendor = {
    name: 'Test Vendor',
    description: 'This is a test vendor',
    features: [
      { name: 'feature 1' },
      { name: 'feature 2' },
      { name: 'feature 3' }
  ]};
  @Input('widget') widget: WidgetComponent
}
