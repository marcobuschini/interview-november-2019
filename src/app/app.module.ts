import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

import { AppComponent } from './app.component';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

import { WidgetModule } from 'widget';
import { MatListModule } from '@angular/material/list';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    WidgetModule,
    MatListModule,
    BrowserAnimationsModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
