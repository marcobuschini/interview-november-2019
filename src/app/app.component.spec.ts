import { TestBed, ComponentFixture, fakeAsync, flush } from '@angular/core/testing';
import { AppComponent } from './app.component';
import { WidgetComponent, Vendor, ParkingSlot } from 'widget';
import { MatCardModule } from '@angular/material/card';
import { MatDividerModule } from '@angular/material/divider';
import { MatListModule } from '@angular/material/list';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { of } from 'rxjs';

describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>;
  let component: AppComponent;
  
  const dummyVendor: Vendor = {
    name: 'Test Vendor',
    description: 'This is a test vendor',
    features: [
      { name: 'Feature 1' },
      { name: 'Feature 2' },
      { name: 'Feature 3' }
    ]
  };

  const dummySlots: ParkingSlot[] = [
    {
      id: 0,
      name: 'SLot 1',
      features: [
        'Feature 1',
        'Feature 2'
      ]
    }
  ]

  beforeEach(async () => {
    TestBed.configureTestingModule({
      declarations: [
        AppComponent,
        WidgetComponent,
      ],
      imports: [
        MatCardModule,
        MatDividerModule,
        MatListModule,
        HttpClientTestingModule,
      ],
    }).compileComponents();

    fixture = TestBed.createComponent(AppComponent);
    fixture.componentInstance.widget = TestBed.createComponent(WidgetComponent).componentInstance;

    let vendorSpy = jest.spyOn(fixture.componentInstance.widget.service, 'getVendorFeatures')
      .mockReturnValue(of(dummyVendor.features));
    let slotsSpy = jest.spyOn(fixture.componentInstance.widget.service, 'getParkingSlots')
      .mockReturnValue(of(dummySlots));

    component = fixture.componentInstance;
    component.widget.ngOnInit();
    fixture.detectChanges();
  });

  afterEach(() => {
    component.widget.ngOnDestroy();
  });

  it('should create', fakeAsync(() => {
    expect(component).toBeTruthy();

    flush();

    expect(fixture).toMatchSnapshot();
  }));
});
