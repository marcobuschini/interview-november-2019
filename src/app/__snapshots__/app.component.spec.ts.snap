// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should create 1`] = `
<app-root
  title={[Function String]}
  vendor={[Function Object]}
  widget={[Function WidgetComponent]}
>
  <parking-widget
    _nghost-a-c0=""
    ng-reflect-vendor="[object Object]"
  >
    <mat-card
      _ngcontent-a-c0=""
      class="mat-card"
    >
      <mat-list
        _ngcontent-a-c0=""
        class="mat-list mat-list-base"
        role="list"
      >
        
        <mat-list-item
          _ngcontent-a-c0=""
          class="mat-list-item"
          role="listitem"
        >
          <div
            class="mat-list-item-content"
          >
            <div
              class="mat-list-item-ripple mat-ripple"
              mat-ripple=""
              ng-reflect-disabled="true"
              ng-reflect-trigger="[object HTMLUnknownElement]"
            />
            <div
              class="mat-list-text"
            />
            Feature 1 
          </div>
        </mat-list-item>
        <mat-list-item
          _ngcontent-a-c0=""
          class="mat-list-item"
          role="listitem"
        >
          <div
            class="mat-list-item-content"
          >
            <div
              class="mat-list-item-ripple mat-ripple"
              mat-ripple=""
              ng-reflect-disabled="true"
              ng-reflect-trigger="[object HTMLUnknownElement]"
            />
            <div
              class="mat-list-text"
            />
            Feature 2 
          </div>
        </mat-list-item>
        <mat-list-item
          _ngcontent-a-c0=""
          class="mat-list-item"
          role="listitem"
        >
          <div
            class="mat-list-item-content"
          >
            <div
              class="mat-list-item-ripple mat-ripple"
              mat-ripple=""
              ng-reflect-disabled="true"
              ng-reflect-trigger="[object HTMLUnknownElement]"
            />
            <div
              class="mat-list-text"
            />
            Feature 3 
          </div>
        </mat-list-item>
      </mat-list>
      <mat-divider
        _ngcontent-a-c0=""
        aria-orientation="horizontal"
        class="mat-divider mat-divider-horizontal"
        role="separator"
      />
      <div
        _ngcontent-a-c0=""
        fxlayout="row"
        fxlayout.xs="column"
        fxlayoutalign="space-between"
      >
        
        <div
          _ngcontent-a-c0=""
        >
           SLot 1 
          <mat-list
            _ngcontent-a-c0=""
            class="mat-list mat-list-base"
            role="list"
          >
            
            <mat-list-item
              _ngcontent-a-c0=""
              class="mat-list-item"
              role="listitem"
            >
              <div
                class="mat-list-item-content"
              >
                <div
                  class="mat-list-item-ripple mat-ripple"
                  mat-ripple=""
                  ng-reflect-disabled="true"
                  ng-reflect-trigger="[object HTMLUnknownElement]"
                />
                <div
                  class="mat-list-text"
                />
                Feature 1 
              </div>
            </mat-list-item>
            <mat-list-item
              _ngcontent-a-c0=""
              class="mat-list-item"
              role="listitem"
            >
              <div
                class="mat-list-item-content"
              >
                <div
                  class="mat-list-item-ripple mat-ripple"
                  mat-ripple=""
                  ng-reflect-disabled="true"
                  ng-reflect-trigger="[object HTMLUnknownElement]"
                />
                <div
                  class="mat-list-text"
                />
                Feature 2 
              </div>
            </mat-list-item>
          </mat-list>
        </div>
      </div>
    </mat-card>
  </parking-widget>
</app-root>
`;
